framework/src/vectorpostprocessors/ElementVariablesDifferenceMax.C

```diff
@@ -0,0 +1,143 @@
+    _b_value(declareVector(getVar("compare_b", 0)->name())),
+    _max_difference(declareVector("Difference")),
+    _position_x(declareVector("X")),
+    _position_y(declareVector("Y")),
+    _position_z(declareVector("Z")),
+    _furthest_from_zero(getParam<bool>("furthest_from_zero")),
+    _preserve_sign(_furthest_from_zero && getParam<bool>("preserve_sign"))
+{
+  // These are all single-value arrays
+  _a_value.resize(1);
+  _b_value.resize(1);
+  _max_difference.resize(1);
+  _position_x.resize(1);
+  _position_y.resize(1);
+  _position_z.resize(1);
+
+  if (getParam<bool>("preserve_sign") != _preserve_sign)
+    mooseDoOnce(
+        mooseWarning("================================ ",
+                     name(),
+                     " ================================\n",
+                     "  block = VectorPostprocessors\n",
+                     "  type = ElementVariablesDifferencesMax\n",
+                     "\n",
+                     "\"preserve_sign\" was set to \"true\" but \"furthest_from_zero\" is not.\n",
+                     "This option will be ignored.\n",
+                     "================================ ",
+                     name(),
+                     " ================================"));
+}
+
+void
+ElementVariablesDifferenceMax::execute()
+{
+  for (unsigned int qp = 0; qp < _qrule->n_points(); ++qp)
+  {
+    // Get the difference and its sign
+    Real difference = _a[qp] - _b[qp];
+    bool is_negative = (difference < 0);
+
+    // Dirty little absolute function
+    if (_furthest_from_zero)
+      difference = is_negative ? -difference : difference;
+
+    // Assign the appropriate values if a new maximum is found
+    if (difference > _max_difference[0])
+    {
+      _max_difference[0] = difference;
+      _max_difference_is_negative = is_negative;
+      _a_value[0] = _a[qp];
+      _b_value[0] = _b[qp];
+
+      _position_x[0] = _q_point[qp](0);
+      _position_y[0] = _q_point[qp](1);
+      _position_z[0] = _q_point[qp](2);
+    }
+  }
+}
+
+void
+ElementVariablesDifferenceMax::finalize()
+{
+  // Gather all the parameters
+  gatherProxyValueMax(_max_difference[0], _a_value[0]);
+  gatherProxyValueMax(_max_difference[0], _b_value[0]);
+  gatherProxyValueMax(_max_difference[0], _max_difference[0]);
+  gatherProxyValueMax(_max_difference[0], _max_difference_is_negative);
+  gatherProxyValueMax(_max_difference[0], _position_x[0]);
+  gatherProxyValueMax(_max_difference[0], _position_y[0]);
+  gatherProxyValueMax(_max_difference[0], _position_z[0]);
+  gatherMax(_max_difference[0]);
+
+  /* Convert back to a negative value if required by _preserve_sign
+   *
+   * The sign of the maximum difference is only swapped when _furthest_from_zero
+   * is true, hence the (otherwise redundant) parenthesization of the last two
+   * terms to make a logical grouping clear to those who read this code
+   */
+  if (_preserve_sign && (_furthest_from_zero && _max_difference_is_negative))
```

Comment:
I believe so, as each provides useful information. I tried to make it generic so others can get the functionality they need, but want the default values to work out-of-the-box according to what most people would expect.

My specific use cases require both the largest absolute value and the largest values. The "preserve" functionality preserves information for those who care about the directionality of the difference.

The values reported by this code should be:

### All Negative
{-5, -3, -1} | Furthest from Zero: T | : F
---- | ---- | ----
Preserve Sign: T | -5 | 1
: F | 5 | 1

### Mixed, Larger Absolute in Negative
{-5, -3, -1, 3, 4} | Furthest from Zero: T | : F
---- | ---- | ----
Preserve Sign: T | -5 | 4
: F | 5 | 4

### Mixed, Large Absolute in Positive
{-4, -3, 1, 3, 5} | Furthest from Zero: T | : F
---- | ---- | ----
Preserve Sign: T | 5 | 5
: F | 5 | 5

### All Positive
{1, 3, 5} | Furthest from Zero: T | : F
---- | ---- | ----
Preserve Sign: T | 5 | 5
: F | 5 | 5

If you really are against all these options I will pull them out, as it may still be useful to others.

Otherwise, based on the "All Negative" Case I realized that it is useful to allow the combination (Furthest = F, Preserve = T), so I will change the code to allow that.